Configuration/NativeMethods.cs

```diff
@@ -64,7 +64,9 @@ private struct AssemblyInfo
         public static string GetAssemblyPath(string name)
         {
             if (name == null)
-                throw new ArgumentNullException("name");
+                throw new ArgumentNullException("name");
+              
+            if( Type.GetType( "Mono.Runtime" ) != null ) return null;
 
```

Comment:
This makes sense. It would be best if it's always safe to create `AzureApplicationSettings` in any environment. `NativeMethods` should generally be considered an implementation detail of `AzureApplicationSettings`. Making it a private nested class may be an additional way to make this clearer and safer.